packages/web/src/utils/pool-utils.ts

```diff
@@ -48,3 +55,55 @@ export function checkPoolStakingRewards(incentivized?: boolean) {
 export function isOrderedTokenPaths(tokenAPath: string, tokenBPath: string): boolean {
   return [tokenAPath, tokenBPath].sort(sortTokenPaths)?.[0] === tokenAPath;
 }
+
+export function sortTokensByPoolOrder<T extends { token: { path: string } }>(
+  items: T[],
+  tokenAPath: string,
+  tokenBPath: string,
+): T[] {
+  const tokenACheckedPath = checkGnotPath(tokenAPath);
+  const tokenBCheckedPath = checkGnotPath(tokenBPath);
+
+  return [...items].sort((a, b) => {
+    const itemACheckedPath = checkGnotPath(a.token.path);
+    const itemBCheckedPath = checkGnotPath(b.token.path);
+
+    const groupA = itemACheckedPath === tokenACheckedPath ? 0 : itemACheckedPath === tokenBCheckedPath ? 1 : 2;
+    const groupB = itemBCheckedPath === tokenACheckedPath ? 0 : itemBCheckedPath === tokenBCheckedPath ? 1 : 2;
```

Comment:
for more readability, it seems good to make as an function.

```suggestion
    const getGroup = (path: string): number => {
      if (path === tokenACheckedPath) return 0;
      if (path === tokenBCheckedPath) return 1;
      return 2;
    }

    const groupA = getGroup(itemACheckedPath);
    const groupB = getGroup(itemBCheckedPath);
```

Reply:
Thanks, We'll refactor it later!